Eviction Thresholds Update

dashpole opened the pull request:
Sets the defaults for the eviction-hard threshold for GCE based on what we were using during testing: "memory.available<250Mi,nodefs.available<10%,nodefs.inodesFree<5%".
Sets flags for e2e tests to use eviction-minimum-reclaim: "nodefs.available<5%,nodefs.inodesFree<5%"

this fixes #32537

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36911)
<!-- Reviewable:end -->

dashpole commented:
wait on this until I check out what the GKE defaults are.

derekwaynecarr commented on pkg/apis/componentconfig/v1alpha1/defaults.go:
```diff
@@ -368,9 +368,12 @@ func SetDefaults_KubeletConfiguration(obj *KubeletConfiguration) {
 	}
 	if obj.EvictionHard == nil {
-		temp := "memory.available<100Mi"
+		temp := "memory.available<300Mi,nodefs.available<10%,nodefs.inodesFree<5%"
```
how was this reached?  i left kubecon thinking we did not have good defaults in mind?

@sjenning

dashpole replied:
This is what we have been using for e2e and e2e_node tests

derekwaynecarr commented:
I think we should separate a GKE / GCE default separate from the open source default?

dashpole commented:
Sorry, this isnt gce/gke specific at all.  Because of #33444, we added some special handling for these flags in GKE, and I wanted to take a look at that before merging.
These values simply are what we have been using on e2e tests.

dashpole commented:
@k8s-bot kubemark e2e test this

derekwaynecarr commented:
Moving the defaults for node_e2e to be the defaults for the kubelet itself seems like a decision that needs to be well-reasoned.  Those defaults are maybe appropriate for node_e2e workload, but they are not really defaults you would want in production.  

For example, doing any default for disk resources should also have the minimum reclaim value set for those values.  I thought we would roll out defaults for disk once we removed the image-gc thresholds.  Until then, I thought we were saying disk eviction is opt-in.

vishh commented:
> For example, doing any default for disk resources should also have the minimum reclaim value set for those values. I thought we would roll out defaults for disk once we removed the image-gc thresholds. Until then, I thought we were saying disk eviction is opt-in.

Good catch. We should remove image-gc thresholds right away. Disk evictions will be useful for almost all users since disk usage often bricks a node. So it might be useful to start with some conservative defaults that do not break existing users. We can continue refining the defaults as we gain more signals from production.

dchen1107 commented:
+1 on @derekwaynecarr above. The Kubelet provides knobs to configure the node, including the resource management, but the value for those knobs should be subjected to the cluster admin. One set of values cannot work for everyone. For example, if the node is using the devicemapper as the graph driver, the configuration of nodefs.inodesFree<5% doesn't matter much here.

dashpole commented:
Eviction flags were already being used on GCE, but I changed them to match what we were using in our tests.  I also added minimum reclaim flags to our testing so that we can start running with that enabled.  This PR now makes no change to the kubelet defualts.

dashpole committed:
added eviction minimum reclaim flags to test flags, and changed gce default config for eviction-hard to match what tests are using

dashpole commented:
@derekwaynecarr @vishh @dchen1107 Do these changes sound good?  They only affect GCE defaults, and test defaults, and do not modify the kubelet defaults.

derekwaynecarr commented:
I am going to remove myself as an approver since I cannot speak for GCE.

I have allocated time for 1.6 to help inform what defaults we would like in open source project.  For the e2e tests, it looks fine to me.

mtaufen commented:
This LGTM
